server/modules/elastic/elasticassistantstore.go

```diff
@@ -0,0 +1,533 @@
+					if source, ok := hit["_source"].(map[string]any); ok {
+						if chat, ok := source[store.schemaPrefix+"chat"].(map[string]any); ok {
+							// Convert the source to a StoredMessage
+							sourceJSON, err := json.Marshal(chat)
+							if err != nil {
+								logger.WithError(err).Error("Failed to marshal message source")
+								continue
+							}
+
+							var message model.StoredMessage
+							if err := json.Unmarshal(sourceJSON, &message); err != nil {
+								logger.WithError(err).Error("Failed to unmarshal StoredMessage")
+								continue
+							}
+
+							message.Auditable.Kind = source[store.schemaPrefix+"kind"].(string)
+							message.Auditable.Id = hit["_id"].(string)
+
+							messages = append(messages, &message)
+						}
+					}
+				}
+			}
+		}
+	}
+
+	logger.WithFields(log.Fields{
+		"messageCount": len(messages),
+		"sessionId":    sessionId,
+		"requestId":    ctx.Value(web.ContextKeyRequestId),
+	}).Debug("Found chat history messages")
+
+	return messages, nil
+}
+
+func (store *ElasticAssistantstore) GetSessions(ctx context.Context, userId string) ([]*model.AssistantSession, error) {
+	if err := store.server.CheckAuthorized(ctx, "read_authored", "assistant"); err != nil {
+		return nil, err
+	}
+
+	logger := log.FromContext(ctx)
+
+	query := map[string]any{
+		"query": map[string]any{
+			"bool": map[string]any{
+				"must": []any{
+					map[string]any{
+						"term": map[string]any{
+							store.schemaPrefix + "session.userId": userId,
+						},
+					},
+					map[string]any{
+						"term": map[string]any{
+							store.schemaPrefix + "kind": "session",
+						},
+					},
+				},
+				"must_not": []any{
+					map[string]any{
+						"exists": map[string]any{
+							"field": store.schemaPrefix + "session.deleteTime",
+						},
+					},
+				},
+			},
+		},
+		"sort": []any{
+			map[string]any{
+				"@timestamp": map[string]any{
+					"order": "asc",
+				},
+			},
+		},
+		"size": 10000,
+	}
+
+	// Convert query to JSON
+	queryJSON, err := json.Marshal(query)
+	if err != nil {
+		logger.WithError(err).Error("Failed to marshal Elasticsearch query")
+		return nil, err
+	}
+
+	logger.WithFields(log.Fields{
+		"query":     store.truncate(string(queryJSON)),
+		"requestId": ctx.Value(web.ContextKeyRequestId),
+	}).Debug("Searching for first messages of each session")
+
+	// Execute search
+	res, err := store.esClient.Search(
+		store.esClient.Search.WithContext(ctx),
+		store.esClient.Search.WithIndex(store.sessionIndex),
+		store.esClient.Search.WithBody(strings.NewReader(string(queryJSON))),
+	)
+	if err != nil {
+		logger.WithError(err).Error("Failed to execute Elasticsearch search")
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	// Read response
+	responseJSON, err := readJsonFromResponse(res)
+	if err != nil {
+		logger.WithError(err).Error("Failed to read Elasticsearch response")
+		return nil, err
+	}
+
+	logger.WithFields(log.Fields{
+		"response":  store.truncate(responseJSON),
```

Comment:
Similar potential log sensitivity issue (and response key could be made more specific)